src/components/Table.jsx

```diff
@@ -0,0 +1,108 @@
+import React, { Component } from 'react'
+import contacts from '../contacts.json';
+import Button from './Button';
+import './Table.css'
+
+
+class Table extends Component {
+    constructor(props) {
+         super(props)
+         this.allContacts = JSON.parse(JSON.stringify(contacts));
+        const firstFiveContacts = this.allContacts.splice(0, 5)
+         this.state = {
+           contacts: firstFiveContacts,
+         };
+        
+}
+
+    contactList() {
+        return this.state.contacts.map(contact => {
+            return (
+                <tr key={contact.id}>
+                    <td> <img className="contactPic" src={contact.pictureUrl} alt=""/></td>
+                    <td>{contact.name}</td>
+                    <td>{contact.popularity.toFixed(2)}</td>
+                    <td><Button key={contact.id} clickEvent={() => this.deleteContact(contact.id)} clase="btn btn-danger">Delete</Button></td>
+                </tr>
+            )
+        })
+    }
+
+    addRandom() {
+        const allContacts = this.allContacts;
+        const contactsCopy = [...this.state.contacts]
+        const getRandom = () => {
+            for (let i = 0; i < contactsCopy.length; i++){
+                const random = allContacts[Math.floor(Math.random() * allContacts.length)]
+                console.log(random.id)
+                const isEqual = contactsCopy.find(elem => elem.id === random.id)
+                if(!isEqual){
+                    return random
+                }
+            }
+        }
+        const randomElement = contactsCopy.length > 0 ? getRandom() : allContacts[Math.floor(Math.random() * allContacts.length)]
+        contactsCopy.push(randomElement)
+        const randomIndex = contactsCopy.findIndex(contact => contact.id === randomElement.id)
+        this.allContacts.splice(randomIndex,1)
+        this.setState({
+            contacts: contactsCopy
+        })
+    }
+
+    sortByName() {
+        const contactsCopy = [...this.state.contacts]
+        const sortedArray = contactsCopy.sort((a,b) => a.name.localeCompare(b.name))
+        this.setState({
+            contacts: sortedArray
+        })
+    }
+
+    sortByPopularity() {
+        const contactsCopy = [...this.state.contacts]
+        const sortedArray = contactsCopy.sort((a, b) => (a.popularity < b.popularity) ? 1 : ((b.popularity < a.popularity) ? -1 : 0))
```

Comment:
`const sortedArray = [...this.state.contacts].sort((a,b) => a.popularity - b.popularity)`